### PR TITLE
fix #1530

### LINKED ISSUE #1530: cache interface: Put(k, v, timeout) the time should the type of time.Duration
RT

### PR DESCRIPTION
**Pull request by youngsterxyf**
@JessonChan @astaxie Please help to review my pr, Thanks!

### CODE REVIEW

File: cache/README.md [hunk 1]
```diff
@@ -26,7 +26,7 @@ Then init a Cache (example with memory adapter)
 
 Use it like this:	
 	
-	bm.Put("astaxie", 1, 10)
+	bm.Put("astaxie", 1, time.Second * 10)
```
Comment (JessonChan) [inline cache/README.md]: why not 10*time.Second ? more readable.

File: cache/memcache/memcache.go [hunk 2]
```diff
@@ -99,7 +100,7 @@ func (rc *Cache) Put(key string, val interface{}, timeout int64) error {
 	if !ok {
 		return errors.New("val must string")
 	}
-	item := memcache.Item{Key: key, Value: []byte(v), Expiration: int32(timeout)}
+	item := memcache.Item{Key: key, Value: []byte(v), Expiration: int32(timeout.Seconds())}
```
Comment (JessonChan) [inline cache/memcache/memcache.go]: timeout/time.Second is more efficient

**Comment by JessonChan** [comment]
@astaxie 
maybe this pull request should be merged after 1.6 releases out?

it's not compatible with beego 1.5 or below.

File: cache/memcache/memcache.go (hunk 2 above)
Comment (youngsterxyf) [inline cache/memcache/memcache.go]: @JessonChan Yes, you are right

File: cache/file.go [hunk 3]
```diff
@@ -142,16 +142,16 @@ func (fc *FileCache) GetMulti(keys []string) []interface{} {
...
 // timeout means how long to keep this file, unit of ms.
 // if timeout equals FileCacheEmbedExpiry(default is 0), cache this item forever.
-func (fc *FileCache) Put(key string, val interface{}, timeout int64) error {
+func (fc *FileCache) Put(key string, val interface{}, timeout time.Duration) error {
 	gob.Register(val)
 
 	item := FileCacheItem{Data: val}
 	if timeout == FileCacheEmbedExpiry {
-		item.Expired = time.Now().Unix() + (86400 * 365 * 10) // ten years
+		item.Expired = time.Now().Add((86400 * 365 * 10) * time.Second) // ten years
```
Comment (JessonChan) [inline cache/file.go]: (24*365*10)*time.Hour? will be more readable

**Comment by JessonChan** [comment]
LGTM  @astaxie

**Pull request merged**